src/main/java/homework10/MyHashMap.java

```diff
@@ -0,0 +1,148 @@
+package homework10;
+
+import java.util.Map;
+
+public class MyHashMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    Node<K, V>[] arrayOfNodes;
+    int size;
+
+    public MyHashMap() {
+    }
+
+    public V get(K key) {
+        return getNode(key).getValue();
+    }
+
+    public V put(K key, V value) {
+        if ((arrayOfNodes == null || arrayOfNodes.length == 0) ^ size == arrayOfNodes.length * LOAD_FACTOR) {
+            arrayOfNodes = resize(arrayOfNodes);
+        }
+        Node<K, V> positionNode = arrayOfNodes[hash(key) % arrayOfNodes.length];
+        if (positionNode == null) {
+            arrayOfNodes[hash(key) % arrayOfNodes.length] = new Node<>(key, value, null);
+            size++;
+        } else if (positionNode.getKey().equals(key) && positionNode.hash == hash(key)) {
+            V oldValue = positionNode.getValue();
+            arrayOfNodes[hash(key) % arrayOfNodes.length].setValue(value);
+            return oldValue;
```

Comment:
we can write:
```suggestion
            return value;
```
and remove: V oldValue = positionNode.getValue();

Reply:
we can't because method returns old value of node =)